File: relisoft-hr/server/models/Holiday.js
import mongoose from 'mongoose';

const holidaySchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    date: { type: Date, required: true },
    day: { type: String },
    type: {
      type: String,
      enum: ['national', 'festival', 'optional'],
      required: true,
    },
    location: [{ type: String }],
    year: { type: Number },
    description: { type: String },
    status: {
      type: String,
      enum: ['active', 'inactive'],
      default: 'active',
    },
  },
  { timestamps: true }
);

export default mongoose.model('Holiday', holidaySchema);

File: relisoft-hr/server/Models/Holiday.cs
using System.ComponentModel.DataAnnotations;

namespace RelisoftHR.Models;

public class Holiday
{
    [Key]
    public int Id { get; set; }
    [Required, MaxLength(200)]
    public string Name { get; set; } = "";
    public DateOnly Date { get; set; }
    [Required, MaxLength(50)]
    public string Type { get; set; } = ""; // "Fixed" or "Optional"
}
